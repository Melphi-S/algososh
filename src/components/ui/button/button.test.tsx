import { Button } from "./button";
import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Button", () => {
  beforeEach(cleanup);

  it("should render with text", function () {
    render(<Button text="test" />);
    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("test");
    expect(button).toMatchSnapshot();
  });

  it("should render without text", function () {
    render(<Button />);
    const button = screen.getByTestId("button");
    expect(button).toHaveTextContent("");
    expect(button).toMatchSnapshot();
  });

  it("should render disabled", function () {
    render(<Button disabled />);
    const button = screen.getByTestId("button");
    expect(button).toBeDisabled();
    expect(button).toMatchSnapshot();
  });

  it("should render with loader", function () {
    render(<Button isLoader />);
    const button = screen.getByTestId("button");
    const loader = screen.getByTestId("loader");
    expect(button).toContainElement(loader);
    expect(button).toMatchSnapshot();
  });

  it("should call callback correctly", function () {
    const mock = jest.fn();
    render(<Button onClick={mock} />);
    const button = screen.getByTestId("button");
    userEvent.click(button);
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
