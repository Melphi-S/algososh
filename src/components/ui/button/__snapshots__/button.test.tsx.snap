// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render disabled 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button should render with loader 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  data-testid="button"
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    data-testid="loader"
    src="loader.svg"
  />
</button>
`;

exports[`Button should render with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  >
    test
  </p>
</button>
`;

exports[`Button should render without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  data-testid="button"
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
